### PR TITLE
GD-198: Add `set_property` to SceneRunner.

### PR DESCRIPTION
**Pull request by clemens-tolboom**
Add `set_property` to both GdUnitSceneRunner and GdUnitSceneRunnerImpl

### CODE REVIEW

**Comment by clemens-tolboom** [comment]
This is a WIP as I added property caching (probably too early and maybe unneeded).

```gdscript
var __properties:Dictionary # <==== cache properties

func set_property(name :String, value) -> void:
	if __properties.size() == 0:
		_current_scene.get_property_list().map(func(i:Dictionary) : __properties[i.name] = i)

	if __properties.has(name):
		_current_scene.set(name, value)
```

**Comment by MikeSchulze** [comment]
Thank you for finding this, 
you can work via `runner.scene().set("my_value", 10)`.
But yes for completeness sake we should implement a set_property.

File: addons/gdUnit4/src/core/GdUnitSceneRunnerImpl.gd [hunk 1]
```diff
@@ -247,6 +247,15 @@ func get_property(name :String):
...
 
+var __properties:Dictionary
+
+func set_property(name :String, value) -> void:
+	if __properties.size() == 0:
+		_current_scene.get_property_list().map(func(i:Dictionary) : __properties[i.name] = i)
+
+	if __properties.has(name):
+		_current_scene.set(name, value)
+
```
Comment (MikeSchulze) [inline addons/gdUnit4/src/core/GdUnitSceneRunnerImpl.gd:258]:
```suggestion
func set_property(name :String, value :Variant) -> bool:
	if _current_scene.get(name) == null:
		push_error("The property named '%s' cannot be set, it does not exist!" % name)
		return false;
	_current_scene.set(name, value)
	return true

```

File: addons/gdUnit4/src/GdUnitSceneRunner.gd [hunk 2]
```diff
@@ -168,6 +168,11 @@ func maximize_view() -> GdUnitSceneRunner:
 func get_property(name :String) -> Variant:
 	return null
 
+## Set the  value <value> of the property with the name <name>.[br]
+## [member name] : name of property[br]
+## [member value] : value of property
+func set_property(name :String, value) -> void:
```
Comment (MikeSchulze) [inline addons/gdUnit4/src/GdUnitSceneRunner.gd:174]:
```suggestion
func set_property(name :String, value :Variant) -> bool:
```

**Review by MikeSchulze — CHANGES_REQUESTED** [review]
Please add also an unit test to `res://addons/gdUnit4/test/core/GdUnitSceneRunnerTest.gd`

**Comment by clemens-tolboom** [comment]
Superseded by #202 ... closing.

**Pull request closed without merging**